src/osd/PG.h

```diff
@@ -314,18 +315,34 @@ class PG : public DoutPrefixProvider {
   ZTracer::Endpoint trace_endpoint;
 
   void lock_suspend_timeout(ThreadPool::TPHandle &handle);
-  void lock(bool no_lockdep = false) const;
+  void lock(bool no_lockdep = false) const {
+    _lock.Lock(no_lockdep);
+	// if we have unrecorded dirty state with the lock dropped, there is a bug
```

Comment:
Isn't this no longer true? So the assertions need to be updated to deal with the storelock.